### PR TITLE
Make non-empty case statements have break or return

### LINKED ISSUE #1044: glsl non-empty case statements must have break or return for ANGLE
ps_border.fs.glsl and ps_border.vs.glsl has non-empty case statements must have break or return, it causes failures of compiling the shaders. I faced the problem on gecko webrender on graphics branch on Windows. When I tested on windows, "discard;"  also seemed to request break/return

The following are actual error log.
>C:\fakepath(4586,1): error X3533: non-empty case statements must have break or return
>C:\fakepath(4589,1): error X3533: non-empty case statements must have break or return

### PR DESCRIPTION
**Pull request by sotaroikeda**
Fixes  #1044

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1054)

### CODE REVIEW

**Comment by emilio** [comment]
Ugh, this looks pretty awful IMO, there's a lot of duplicated code. Are you sure just `break`ing under the `default` cases doesn't fix it?

**Comment by glennw** [comment]
Perhaps we can at least put the duplicated code sections into functions, so that only the function call is duplicated?

**Comment by sotaroikeda** [comment]
I created the pull request be referencing #749. But I misunderstood the error code, I tried to minimize the change, and duplicated code parts were not necessary.

**Comment by sotaroikeda** [comment]
> Are you sure just breaking under the default cases doesn't fix it?

It seems to work, I am going to update the pull request.

File: webrender/res/ps_border.fs.glsl [hunk 1]
```diff
@@ -458,10 +459,8 @@ void main(void) {
         case BORDER_STYLE_RIDGE:
           draw_mixed_border(distance_from_mix_line, distance_from_middle, local_pos, brightness_mod.xy);
           break;
-        case BORDER_STYLE_HIDDEN:
-          discard;
         default:
-          discard;
+          break;
```
Comment (kvark) [inline webrender/res/ps_border.fs.glsl]: removing `discard` here will change the behavior

**Comment by kvark** [comment]
@bors-servo r-
sorry for interrupting - I'd like the `discard` changes to be clarified

**Comment by glennw** [comment]
@kvark Ah, you're right - I misread that change, thinking that the discard was added for the hidden case.

**Comment by sotaroikeda** [comment]
I removed discard since the following did not work on windows
>        case BORDER_STYLE_HIDDEN:
>          discard;

The following works on windows, but it did not work on linux
>        case BORDER_STYLE_HIDDEN:
>          discard;
>          break;

**Comment by sotaroikeda** [comment]
> removing discard here will change the behavior

Hmm

**Comment by kvark** [comment]
Try working around it by moving the discard to the end of an execution path (e.g. an "if" branch).

**Pull request merged**